frontend/src/lib/components/PropertyFilters/PropertyFilter.js

```diff
@@ -110,24 +125,105 @@ export function PropertyFilter({ index, onComplete, logic }) {
                     </Select>
                 </div>
             )}
-            {key && !isOperatorFlag(operator) && (
-                <div className="col-5 pl-0">
+            {displayOperatorAndValue && !isOperatorFlag(operator) && (
+                <div className="col-5 p-0">
                     <PropertyValue
                         type={type}
-                        key={key}
-                        propertyKey={key}
+                        key={propkey}
+                        propertyKey={propkey}
                         operator={operator}
                         value={value}
                         onSet={(value) => {
                             onComplete()
-                            setFilter(index, key, value, operator, type)
+                            setThisFilter(propkey, value, operator, type)
                         }}
                     />
                     {(operator === 'gt' || operator === 'lt') && isNaN(value) && (
                         <p className="text-danger">Value needs to be a number. Try "equals" or "contains" instead.</p>
                     )}
                 </div>
             )}
-        </div>
+        </>
+    )
+}
+
+function CohortPaneContents({ onComplete, setThisFilter, propkey, displayOperatorAndValue }) {
+    const { cohorts } = useValues(cohortsModel)
+
+    return (
+        <>
+            <Select
+                style={{ width: '100%' }}
+                labelInValue
+                placeholder="Cohort name"
+                value={
+                    displayOperatorAndValue
+                        ? { value: '' }
+                        : {
+                              value: propkey,
+                              label:
+                                  cohorts?.find((cohort) => cohort.id === parseInt(propkey?.substring(1)))?.name ||
+                                  propkey,
+                          }
+                }
+                onChange={(_, new_cohort) => {
+                    onComplete()
+                    setThisFilter(new_cohort.value, true, undefined, 'cohort')
+                }}
+            >
+                {cohorts.map((item, index) => (
+                    <Select.Option
+                        key={'cohort_' + item.id}
+                        value={`#${item.id}`}
```

Comment:
This is a little weird and feels a little brittle. Wouldn't it be easier to just have a fixed key `cohort_id`, then make the value the id of the cohort?

Reply:
I think it indeed would. Reworked this type so that it's
```TypeScript
interface CohortProperty {
	key: 'id', // was: string (hashtag followed by cohort ID, e.g. '#42'), now: 'id'
	value: number, // was: true, now: number (cohort ID, e.g. 42)
	type: 'cohort'
}
```
both front and back end.